### PR TITLE
Add triggering of property handler in dbparameters

### PR DESCRIPTION
**Pull request by davidrot**

### CODE REVIEW

File: RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs [hunk 1]
```diff
@@ -372,7 +377,11 @@ public static class DbCommandExtension
             var typeHandler = PropertyHandlerCache.Get<object>(valueType);
             if (typeHandler != null)
             {
-                // TODO: Ensure to reuse the existing PropertyHandler (if given)
+                // It is hard to pre-compile this as the property handler is dynamic
+                var setMethod = typeHandler.GetType().GetMethod("Set");
+                var returnType = setMethod.ReturnType;
+                dbType = m_clientTypeToDbTypeResolver.Resolve(returnType);
+                value = setMethod.Invoke(typeHandler, new[] { value, null });
```
Comment (davidrot) [inline RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: Is the `null` right? @mikependon

Comment (mikependon) [reply RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: It is important to pass the instance of ClassProperty as part of the context of property handler execution.

Do you have the instance of ProperyInfo or even just a name of the property? If yes, then please use the PropertyCache.Get(entityType) to get the list of ClassProperty objects. From there, you can find the equivalent ClassProperty object by iterating the list of properties and by equating the property mapped name to ClassProperty.GetMappedName() method of each instance.

Comment (mikependon) [reply RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: FYI: Use the PropertyMappedNameCache.Get(PropertyInfo) to get the mapped name of the PropertyInfo object.

Comment (davidrot) [reply RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: @mikependon Thank you for your response. Over which properties should it iterate, there is no context or relation to the type.

Comment (mikependon) [reply RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: Good question. Would it be possible for you to pass the related entity type here?

- Via [Dictionary<string, object>](https://github.com/mikependon/RepoDb/blob/5f57e9d9d468b3591029d3c3e3b1ff4a7d8a6891/RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs#L113)
- Via [QueryField](https://github.com/mikependon/RepoDb/blob/5f57e9d9d468b3591029d3c3e3b1ff4a7d8a6891/RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs#L119)
- Via [IEnumerable<QueryField>](https://github.com/mikependon/RepoDb/blob/5f57e9d9d468b3591029d3c3e3b1ff4a7d8a6891/RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs#L125)
- Via [QueryGroup](https://github.com/mikependon/RepoDb/blob/5f57e9d9d468b3591029d3c3e3b1ff4a7d8a6891/RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs#L131)

By moving this [one](https://github.com/mikependon/RepoDb/blob/5f57e9d9d468b3591029d3c3e3b1ff4a7d8a6891/RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs#L137) to the top prior to the calls to the `CreateParameters()` method?

If that is possible, then you can add an additional argument `entityType` in the `CreateParameters()` methods.

Comment (mikependon) [reply RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: Hmmm, upon realizing here, I think I had replied to you wrongly. Let me debug this one now. :)

Comment (mikependon) [reply RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs:384]: What you had made here is correct. For the `QueryField`, `IEnumerable<QueryField>`, `QueryGroup` and `Dictionary<string, object>` parameter that does not have the property property entity type mapping, it should be ignoring the `ClassProperty`. I would humbly say that only the type level must be handled - specifically on this case.

**Comment by mikependon** [comment]
After review, what you have done is correct. Thank you for this PR. Cheers!

**Comment by mikependon** [comment]
@davidrot - I actually added more Unit Tests for this critical behavior. Also, I added the `entityType` argument in the `DbCommandExtensions.CreateParameters()` method.

**Pull request merged**